File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
 
        char_table = [0] * 26

        for i in s:
            char_table[ord(i) - ord('a')] += 1

        for i in t:
            char_table[ord(i) - ord('a')] -= 1 
            if char_table[ord(i) - ord('a')] < 0:
                return False
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()){
            return false;
        }
        else{
        int charPos [] = new int[26];
        for (char c : s.toCharArray()){
            charPos[(int) c - 97] += 1 ;
        }
        for (char c1 : t.toCharArray()){
            if (charPos[(int) c1 - 97] - 1 < 0){
                return false;
            }
            charPos[(int) c1 - 97] -= 1;
        }
        return true;
        }
    }
}